core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -170,6 +171,36 @@ public static boolean hasPlugin(MavenProject project, String plugin) {
         return project.getPlugin(plugin) != null;
     }
 
+    public static boolean hasPluginOfAnyGroupId(MavenProject project, String pluginArtifact) {
+        return getPluginOfAnyGroupId(project, pluginArtifact) != null;
+    }
+
+    public static Plugin getPluginOfAnyGroupId(MavenProject project, String pluginArtifact) {
+        return getPlugin(project, null, pluginArtifact);
+    }
+
+    /**
+     * Returns the plugin with the given groupId (if present) and artifactId (if present).
+     */
+    public static Plugin getPlugin(MavenProject project, String groupId, String artifactId) {
+        if (groupId == null && artifactId == null) {
+            throw new IllegalArgumentException("groupId and artifactId cannot be both null");
+        }
+
+        List<Plugin> plugins = project.getBuildPlugins();
+        if (plugins != null) {
+            for (Plugin plugin : plugins) {
+                boolean matchesArtifactId = artifactId == null || Objects.equal(artifactId, plugin.getArtifactId());
+                boolean matchesGroupId = groupId == null || Objects.equal(groupId, plugin.getGroupId());
```

Comment:
Can we try to avoid usage of fabric8 classes like `Objects` as long as the depedency removal PR is not merged ? This avoid conflicts when rebasing the PR (and also when syncing the 3.5 and 4.0 branch)